src/QuickMarcEditor/useSaveRecord/useSubmitRecord/useSumbitRecord.js

```diff
@@ -96,13 +101,13 @@ const useSubmitRecord = ({
       [MARC_TYPES.HOLDINGS]: `${basePath}/edit-holdings/${externalId}`,
     };
 
-    await refreshPageData(fieldIds, QUICK_MARC_ACTIONS.EDIT, externalId);
-
-    history.push({
+    await history.push({
       pathname: routes[marcType],
       search: searchParams.toString(),
     });
-  }, [basePath, marcType, location, history, refreshPageData, stripes]);
+
+    await refreshPageData(fieldIds, QUICK_MARC_ACTIONS.EDIT, externalId);
```

Comment:
What do you think if we pass the `nextIsShared` argument to the `refreshPageData` function and use it like other `next...` parameters in the `loadData` function? This will work too.

```
const loadData = useCallback(async (fieldIds, nextAction, nextExternalId, nextIsShared) => {
   const _action = nextAction || action;
   const _externalId = nextExternalId || externalId;
   ---> const _isShared = nextIsShared ?? isRequestToCentralTenantFromMember. 

   const headers = _isShared 
      ? { headers: getHeaders(centralTenantId, token, locale) }
      : {};
```

I think it would be easier to maintain a single source of truth, otherwise, there will be two (ref and URL). Even now, there are some cases when the URL isn't in line with the ref.

Reply:
I've changed the approach for this task - now instead of a context there is a `useIsShared` hook that manages the `shared` url parameter.